### PR TITLE
chore: bump deps

### PR DESCRIPTION
**Pull request by He-Pin**

### CODE REVIEW

File: build.mill [hunk 1]
```diff
@@ -98,7 +98,7 @@ object sjsonnet extends VersionFileModule {
   object js extends Cross[SjsonnetJsModule](scalaVersions)
   trait SjsonnetJsModule extends SjsonnetCrossModule with ScalaJSModule with SjsonnetPublishModule {
     def moduleDir = super.moduleDir / os.up
-    def scalaJSVersion = "1.17.0"
+    def scalaJSVersion = "1.20.1"
```
Comment (He-Pin) [inline build.mill:101]: Can we upgrade the scalajs version ? @stephenamar-db

File: sjsonnet/test/src-jvm/sjsonnet/BufferedRandomAccessFileTests.scala [hunk 2]
```diff
@@ -36,7 +36,7 @@ object BufferedRandomAccessFileTests extends TestSuite {
 
       // Boundary conditions
       assert(bufferedFile.readChar(testContent.length - 1) == '.')
-      intercept[IndexOutOfBoundsException] {
+      assertThrows[IndexOutOfBoundsException] {
```
Comment (He-Pin) [inline sjsonnet/test/src-jvm/sjsonnet/BufferedRandomAccessFileTests.scala:39]: new utest requires this
